etc/org.opencastproject.ui.metadata.CatalogUIAdapterFactory-episode-common.cfg

```diff
@@ -144,3 +144,11 @@ property.license.readOnly=false
 property.license.required=false
 property.license.listprovider=LICENSES
 property.license.order=5
+
+# Publisher
+property.publisher.inputID=publisher
+property.publisher.label=EVENTS.EVENTS.DETAILS.METADATA.PUBLISHER
+property.publisher.type=text
+property.publisher.readOnly=true
```

Comment:
Why read only?

Reply:
The publisher is the one who creates the new event, the one that is logged in. Once created, it would not make sense to change it.

Reply:
@jchssystems Our site's use case is that the publisher is the person responsible for the publishing of the event, not necessarily the one performing a manual schedule or a manual upload. Our site requires the one performing the upload or scheduling the event to re-assign the correct point person if the current person is not that person. I have tested your pull and realize that the auto-fill process the current user negates the listprovider. This means the current user cannot get the drop down list of other publishers if current publisher want to change the assignment. I'm trying to find a way to work around that (auto-select or assign the current user from the listprovided set of publishers), but any help you can provide now to make it easier is grateful appreciated!

Reply:
@jchssystems @KatrinIhler, I like this pull,  but I am -1 for making the publisher field unediable by default prior to event publication. I can see that it might be useful to freeze the value of this field after publication.

Reply:
@karendolan Maybe something could be done.

With:
_property.license.readOnly=true_
could work as it does now.

With:
_property.license.readOnly=false_
_property.license.listprovider=PUBLISHER_
could work as you need if I change the code of "new event". You can have a list of all the editors plus the user who is logged in for example.

It would be also possible to have any list at first as provider.
_property.license.listprovider=CONTRIBUTORS_

I have not tested anything yet. Let me take a look and I will tell you something tomorrow.

Reply:
Hi again @karendolan, @KatrinIhler, the code works also with the publisher list provider, but the name is EVENTS.PUBLISHER:

property.publisher.readOnly=false
property.publisher.listprovider=EVENTS.PUBLISHER

With this provider:
1. If there are no publishers yet, the list is empty (?).
2. If there are publishers and the logged in user is already a publisher, the user is selected.
3. If there are publishers and the logged in is not a publisher yet, no-option is selected.

With this provider, the question is, how can an user be a publisher first time? I suggest that, if publisher property is a list, this list be the users of the list provider plus the logged in user. So first time, if an user wants to be a publisher, he must log in and creates an event. Next times, another user can create events and assign him as publisher. What do you think about this solution? Have you another better solution?

Reply:
Hi @karendolan, I have upload a new version of the branch. I think that it is what you need.

By default, the publisher is the logged in user and you cannot change it.

But if you change the file configuration
_etc/org.opencastproject.ui.metadata.CatalogUIAdapterFactory-episode-common.cfg_,
for example with the lines:

property.publisher.readOnly=false
property.publisher.listprovider=PROVIDER

you can choose the publisher between the users of the list. Once the event has been created, this field is read only.

The new event wizard shows all the users of the provider list, plus the logged in user. The logged in user is selected by default.

If you want the list of publishers as provider, you must set:
property.publisher.listprovider=EVENTS.PUBLISHER

The filter of publishers in the events page always shows the users who have published at least one event.

If the list has only one item, no one is selected. There is a bug that I have already reported:
MH-12971-The list of series in the Event Details window does not work properly
I think that it occurs in all lists, when there is only one item.

Please, check the branch if you can and tell me if that is fit to you.

Reply:
Hi @jchssystems Thank you for all the work! I'm trying it all out in the next couple hours...

Reply:
@karendolan,

1. _"If there are no publishers yet, the list contains the currently logged in user and auto-selected to that user."_
Be careful with the bug. If the list has only one item, publisher.setValue(...) does not work. I think It occurs in all lists.

2 _"If the series selection is change again, the field will also be changed again with that series publisher. The user can then override the new change by choosing themself or another from the list."_
But if this occurs before the event is created. After that, the field is read only. If you want to make it editable, you must delete:
newPublisher.setReadOnly(true);
in EventUtils.java
The list in events properties is really there. Only making newPublisher true, the field is read only and you do not see the list.

Reply:
@jchssystems & @KatrinIhler I finally finished tested everything locally and it works great for our use case! Thank you so much for offering this pull.